components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/GroupDAOImpl.java

```diff
@@ -18,188 +18,179 @@
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.Error.UNSUPPORTED_USER_STORE_MANAGER;
+import static org.wso2.carbon.user.core.constants.UserCoreErrorConstants.ErrorMessages.ERROR_NO_GROUP_FOUND_WITH_ID;
+import static org.wso2.carbon.user.core.constants.UserCoreErrorConstants.ErrorMessages.ERROR_NO_GROUP_FOUND_WITH_NAME;
 
 /**
  * RoleDAO Implementation.
  */
 public class GroupDAOImpl implements GroupDAO {
 
+    private static final Log LOG = LogFactory.getLog(GroupDAOImpl.class);
+
     @Override
     public String getGroupNameByID(String id, String tenantDomain) throws IdentityRoleManagementException {
 
-        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        String groupName = null;
-        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
-            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection, GET_GROUP_NAME_BY_ID_SQL)) {
-                statement.setInt(RoleConstants.RoleTableColumns.TENANT_ID, tenantId);
-                statement.setString(RoleConstants.RoleTableColumns.ATTR_NAME, RoleConstants.ID_URI);
-                statement.setString(RoleConstants.RoleTableColumns.ATTR_VALUE, id);
-                int count = 0;
-                try (ResultSet resultSet = statement.executeQuery()) {
-                    while (resultSet.next()) {
-                        // Handle multiple matching groups.
-                        count++;
-                        if (count > 1) {
-                            String errorMessage =
-                                    "Invalid scenario. Multiple groups found for the given group ID: " + id + " and "
-                                            + "tenantDomain: " + tenantDomain;
-                            throw new IdentityRoleManagementClientException(INVALID_REQUEST.getCode(), errorMessage);
-                        }
-                        groupName = resultSet.getString(1);
-                    }
+        try {
+            AbstractUserStoreManager userStoreManager = getUserStoreManager(tenantDomain);
+            return userStoreManager.getGroupNameByGroupId(id);
+        } catch (UserStoreException e) {
+            String errorMessage = String.format("Error while resolving the group name for the given " +
+                            "group ID: %s in tenantDomain: %s", id, tenantDomain);
+            if (e instanceof UserStoreClientException) {
+                // This is to ensure backward compatibility with the previous implementation.
+                if (StringUtils.isNotBlank(e.getErrorCode()) && ERROR_NO_GROUP_FOUND_WITH_ID.getCode()
+                        .equals(e.getErrorCode())) {
+                    LOG.debug(String.format("No group found for the given group ID: %s in the tenantDomain: %s. " +
+                                    "Therefore, returning an empty String.", id, tenantDomain));
+                    return StringUtils.EMPTY;
```

Comment:
Previously, we returned null if no group name was found for the given ID. Why do we return empty here? This may break some flows, right?

Reply:
**Returning Null is not a good practice**. Tested with initial starting sequence and verified no issues by sending an empty string. But I do understand your concern. Will improve.